addons/mail/models/mail_link_preview_message.py

```diff
@@ -0,0 +1,48 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models
+from odoo.addons.mail.tools.discuss import Store
+
+
+class LinkPreviewMessage(models.Model):
+    _name = "mail.link.preview.message"
+    _inherit = "bus.listener.mixin"
+    _description = "Link between link previews and messages"
+
+    link_preview_id = fields.Many2one("mail.link.preview", index=True)
+    message_id = fields.Many2one("mail.message")
```

Comment:
Not sure for this one, is message aways required or could you have link preview "without linked message" ?
```suggestion
    message_id = fields.Many2one("mail.message", required=True, ondelete="cascade")
```

Reply:
You can have a link preview without a message (see usage of `_search_or_create_from_url`)

Reply:
But if we have a link between a link preview and a message, and if we remove the message, shouldn't we remove the link ? The preview stays but the "preview.message" link won't make sense anymore, does it ?

Reply:
I guess it should indeed be deleted both ways